subprojects/dependency-lock/src/main/java/org/gradle/internal/dependencylock/DependencyLockState.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.dependencylock;
+
+import org.gradle.api.artifacts.result.ResolutionResult;
+import org.gradle.internal.dependencylock.model.DependencyLock;
+
+public interface DependencyLockState {
+
+    void resolveAndPersist(String projectPath, String configurationName, ResolutionResult resolutionResult);
```

Comment:
❌ I don't think we should be binding this to projects and configurations, but just have a single "id". This can be the concatenation of project path + configuration name for now, but allows us to add different things later, e.g. locking the classpath of the settings.gradle file.

Reply:
done